### PR TITLE
Add blog for PersistentVolume last phase transition time

### PR DESCRIPTION
**Pull request by RomanBednar**
KEP: https://github.com/kubernetes/enhancements/pull/3796

### CODE REVIEW

**Comment by RomanBednar** [comment]
@jsafrane does this look good you you from technical side?

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+The new LastPhaseTransitionTime field added to the PV status is updated every time a PV transitions from one phase to another.
+Whether it's transitioning from "Pending" to "Bound," "Bound" to "Released," or any other phase transition, the LastPhaseTransitionTime will be recorded.
+For newly created PVs the phase will be set to "Pending" and the LastPhaseTransitionTime will be recorded as well.
+
+This feature allows cluster administrators to:
+
+1. Implement Retention Policies
+
+   With the LastPhaseTransitionTime, administrators can now track when a PV was last used or transitioned to the "Released" phase.
+   This information can be crucial for implementing retention policies to clean up resources that have been in the "Released" phase for a specific duration.
```
Comment (jsafrane) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:34]: Show how crucial it is. "For example, it's trivial to write a script or a policy that deletes all PVs that are in Released phase for 1 week".

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+2. Monitor Storage Health
+
+   By analyzing the phase transition times of PVs, administrators can monitor storage health more effectively.
+   For example, they can identify PVs that have been in the "Pending" phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
+
+3. Produce Alerts and Metrics
+
+   The LastPhaseTransitionTime can be utilized to produce alerts and metrics.
```
Comment (jsafrane) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:43]: If we need such metrics, we need code in KCM / PV controller that would emit them. I'm not sure it's a goal of this feature.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 1 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:34]: Example added.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+layout: blog
+title: "Kubernetes 1.28: Persistent Volume Last Phase Transition Time"
+date: 2023-08-15T10:00:00-08:00
+slug: kubernetes-1-28-persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we are introducing a new alpha feature that aims to improve PersistentVolume (PV) management and help cluster administrators gain better insights into the lifecycle of PVs.
+With the addition of the LastPhaseTransitionTime field in the PV status, administrators will be able to track the last time a PV transitioned to a different phase, allowing for more efficient and informed resource management.
```
Comment (tengqm) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:11]: Please use user facing field names when possible.
This blog is for users, not for developers.
A user doesn't need to know that the field is named as `LastPhaseTransistionTime` in Go lang and the first character must be upper case for it to be treated public.
Instead, he or she need to know the field name as `lastPhaseTransistionTime`.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+   For example, it is now trivial to write a script or a policy that deletes all PVs that have been in "Released" phase for a week.
+
+2. Monitor Storage Health
+
+   By analyzing the phase transition times of PVs, administrators can monitor storage health more effectively.
+   For example, they can identify PVs that have been in the "Pending" phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
+
+## How to use it
+
+The LastPhaseTransitionTime field is currently alpha so it requires `PersistentVolumeLastPhaseTransitionTime` feature gate to be enabled.
```
Comment (tengqm) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:44]:
```suggestion
The LastPhaseTransitionTime field is currently alpha so it requires
the `PersistentVolumeLastPhaseTransitionTime` feature gate to be enabled.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+If you want to test the feature whilst it's alpha, you need to enable this feature gate in the kube-controller-manager and the kube-apiserver.
```
Comment (tengqm) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:46]:
```suggestion
If you want to test the feature whilst it's alpha, you need to enable this feature gate on the `kube-controller-manager` and the `kube-apiserver`.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 3 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: Acknowledged, I've applied your suggestions and replaced all other references to that field to be lower case. Thank you for review.

**Comment by jsafrane** [comment]
lgtm from the sig-storage POV.

**Review by sftim — COMMENTED** [review]
/hold

Release comms would need to assign a publication date for this. OK to unhold once that's happened and the article is updated accordingly.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we are introducing a new alpha feature that aims to improve PersistentVolume (PV) management and help cluster administrators gain better insights into the lifecycle of PVs.
+With the addition of the lastPhaseTransitionTime field in the PV status, administrators will be able to track the last time a PV transitioned to a different phase, allowing for more efficient and informed resource management.
+
+## Why do we need new PV field
+
+PersistentVolumes in Kubernetes play a crucial role in providing storage resources to workloads running in the cluster.
+However, managing these PVs effectively can be challenging, especially when it comes to determining the last time a PV transitioned between different phases, such as "Pending," "Bound," or "Released."
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:16]:
```suggestion
However, managing these PVs effectively can be challenging, especially when it comes to determining the last time a PV transitioned between different phases, such as `Pending`, `Bound` or `Released`.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+Administrators often need to know when a PV was last used or transitioned to certain phases, for instance, to implement retention policies, perform cleanup, or monitor storage health.
+
+In the past, Kubernetes users have faced data loss issues when using the "Delete" retain policy and had to resort to the safer "Retain" policy.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:19]:
```suggestion
In the past, Kubernetes users have faced data loss issues when using the `Delete` retain policy and had to resort to the safer `Retain` policy.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+By introducing the lastPhaseTransitionTime field, we aim to provide a more generic solution that can be used for various use cases,
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:20]:
```suggestion
By introducing the `lastPhaseTransitionTime` field, we aim to provide a more generic solution that can be used for various use cases,
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+including manual cleanup based on the time a volume was last used or producing alerts based on phase transition times.
+
+## How lastPhaseTransitionTime helps
+
+The new lastPhaseTransitionTime field added to the PV status is updated every time a PV transitions from one phase to another.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:25]:
```suggestion
Provided you've enabled the feature gate (see [How to use it](#how-to-use-it), the new `.status.lastPhaseTransitionTime` field of a PersistentVolume (PV)
is updated every time that PV transitions from one phase to another.
``

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+Whether it's transitioning from "Pending" to "Bound," "Bound" to "Released," or any other phase transition, the lastPhaseTransitionTime will be recorded.
+For newly created PVs the phase will be set to "Pending" and the lastPhaseTransitionTime will be recorded as well.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:27]:
```suggestion
Whether it's transitioning from `Pending` to `Bound`, `Bound` to `Released`, or any other phase transition, the `lastPhaseTransitionTime` will be recorded.
For newly created PVs the phase will be set to `Pending` and the `lastPhaseTransitionTime` will be recorded as well.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+This feature allows cluster administrators to:
+
+1. Implement Retention Policies
+
+   With the lastPhaseTransitionTime, administrators can now track when a PV was last used or transitioned to the "Released" phase.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:33]:
```suggestion
   With the `lastPhaseTransitionTime`, administrators can now track when a PV was last used or transitioned to the `Released` phase.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+   This information can be crucial for implementing retention policies to clean up resources that have been in the "Released" phase for a specific duration.
+   For example, it is now trivial to write a script or a policy that deletes all PVs that have been in "Released" phase for a week.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:35]:
```suggestion
   This information can be crucial for implementing retention policies to clean up resources that have been in the `Released` phase for a specific duration.
   For example, it is now trivial to write a script or a policy that deletes all PVs that have been in the `Released` phase for a week.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+2. Monitor Storage Health
+
+   By analyzing the phase transition times of PVs, administrators can monitor storage health more effectively.
+   For example, they can identify PVs that have been in the "Pending" phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:40]:
```suggestion
   For example, they can identify PVs that have been in the `Pending` phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+## How to use it
+
+The lastPhaseTransitionTime field is currently alpha so it requires
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:44]:
```suggestion
The `lastPhaseTransitionTime` field is alpha for Kubernetes v1.28, so it requires
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+
+If you want to test the feature whilst it's alpha, you need to enable this feature gate on the `kube-controller-manager` and the `kube-apiserver`.
+
+Use the `--feature-gates` command line argument:
+
+```
+--feature-gates="...,PersistentVolumeLastPhaseTransitionTime=true"
+```
+
+Keep in mind that the feature enablement does not have immediate effect, the new field will be populated whenever a PV is updated and transitions phases.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:55]:
```suggestion
Keep in mind that the feature enablement does not have immediate effect; the new field will be populated whenever a PV is updated and transitions between phases.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+Administrators can then access the new field through the PV status, which can be retrieved using standard Kubernetes API calls or through Kubernetes client libraries.
+
+Here is an example of how to retrieve the lastPhaseTransitionTime for a specific PV using the kubectl command-line tool:
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:58]:
```suggestion
Here is an example of how to retrieve the `lastPhaseTransitionTime` for a specific PV using the `kubectl` command-line tool:
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+
+```
+kubectl get pv <pv-name> -o jsonpath='{.status.lastPhaseTransitionTime}'
+```
+
+## Going forward
+
+This feature is initially introduced as an alpha feature and will be behind a feature gate.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:66]:
```suggestion
This feature was initially introduced as an alpha feature, behind a feature gate that is disabled by default.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+During the alpha phase, we will collect feedback from the community and address any issues or improvements identified.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:67]:
```suggestion
During the alpha phase, we (Kubernetes SIG Storage) will collect feedback from the end user community and address any issues or improvements identified.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+layout: blog
+title: "Kubernetes 1.28: Persistent Volume Last Phase Transition Time"
+date: 2023-08-15T10:00:00-08:00
+slug: kubernetes-1-28-persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we are introducing a new alpha feature that aims to improve PersistentVolume (PV) management and help cluster administrators gain better insights into the lifecycle of PVs.
+With the addition of the lastPhaseTransitionTime field in the PV status, administrators will be able to track the last time a PV transitioned to a different phase, allowing for more efficient and informed resource management.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:11]:
```suggestion
With the addition of the `lastPhaseTransitionTime` field into the status of a PV, administrators will be able to track the last time a PV transitioned to a different [phase](/docs/concepts/storage/persistent-volumes/#phase), allowing for more efficient and informed resource management.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 20]
```diff
@@ -0,0 +1,88 @@
+---
+layout: blog
+title: "Kubernetes 1.28: Persistent Volume Last Phase Transition Time"
+date: 2023-08-15T10:00:00-08:00
+slug: kubernetes-1-28-persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we are introducing a new alpha feature that aims to improve PersistentVolume (PV) management and help cluster administrators gain better insights into the lifecycle of PVs.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:10]:
```suggestion
In Kubernetes v1.28, we have introduced a new alpha feature that aims to improve PersistentVolume (PV)
storage management and help cluster administrators gain better insights into the lifecycle of PVs.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+With the addition of the lastPhaseTransitionTime field in the PV status, administrators will be able to track the last time a PV transitioned to a different phase, allowing for more efficient and informed resource management.
+
+## Why do we need new PV field
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:13]:
```suggestion
## Why do we need new PV field? {#why-new-field}
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+Once sufficient feedback has been received, or no complaints are received the feature can move to beta.
+The beta phase will allow us to further validate the implementation and ensure its stability.
+At least two Kubernetes releases will be available between the beta phase and the general availability (GA) release.
+
+## Getting involved
+
+We always welcome new contributors so if you would like to get involved you can
+join our [Kubernetes Storage Special-Interest-Group](https://github.com/kubernetes/community/tree/master/sig-storage) (SIG).
+
+If you would like to share feedback, you can do so on our [public Slack channel](https://app.slack.com/client/T09NY5SBT/C09QZFCE5).
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:78]:
```suggestion
If you would like to share feedback, you can do so on our [public Slack channel](https://app.slack.com/client/T09NY5SBT/C09QZFCE5). If you're not already part of that Slack workspace, you can visit https://slack.k8s.io/ for an invitation.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+## Going forward
+
+This feature is initially introduced as an alpha feature and will be behind a feature gate.
+During the alpha phase, we will collect feedback from the community and address any issues or improvements identified.
+
+Once sufficient feedback has been received, or no complaints are received the feature can move to beta.
+The beta phase will allow us to further validate the implementation and ensure its stability.
+At least two Kubernetes releases will be available between the beta phase and the general availability (GA) release.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:71]: Is this true?
```markdown
At least two Kubernetes releases will happen between the release where this field graduates
to beta and the release that graduates the field to general availability (GA). That means that
the earliest release where this field could be generally available is Kubernetes 1.32,
likely to be scheduled for early 2025.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+```
+--feature-gates="...,PersistentVolumeLastPhaseTransitionTime=true"
+```
+
+Keep in mind that the feature enablement does not have immediate effect, the new field will be populated whenever a PV is updated and transitions phases.
+Administrators can then access the new field through the PV status, which can be retrieved using standard Kubernetes API calls or through Kubernetes client libraries.
+
+Here is an example of how to retrieve the lastPhaseTransitionTime for a specific PV using the kubectl command-line tool:
+
+```
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:60]: Optionally: mark this as shell

**Review by sftim — COMMENTED** [review]
Some feedback on the writing

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 23 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: This suggestion is more accurate as there is no guarantee on the release count in beta phase, thanks!

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 24 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: Added.

**Comment by RomanBednar** [comment]
@sftim Seems like the freeze is in effect, any chance to get this merged now?

**Comment by sftim** [comment]
See https://github.com/kubernetes/website/pull/41981#pullrequestreview-1575238861

We'll need release comms to assign a date; I'll nudge that team.

**Comment by bradmccoydev** [comment]
Can we please put Aug 23 publication date?

**Comment by sftim** [comment]
@bradmccoydev do you still want to include this article? We'd need a new date, maybe next week.

**Comment by RomanBednar** [comment]
@bradmccoydev @sftim Any updates on this one?

**Comment by RomanBednar** [comment]
@sftim Sure, does it mean just removing `Kubernetes v1.28` from the heading (and slug), updating date and leaving the rest as it is?

**Comment by sftim** [comment]
> @sftim Sure, does it mean just removing `Kubernetes v1.28` from the heading (and slug), updating date and leaving the rest as it is?

I'm afraid you'll also need to make some further (small) edits to make the article work right as a standalone piece. If you're not sure what to do, please make a start and a reviewer can provide feedback.

**Comment by RomanBednar** [comment]
> I'm afraid you'll also need to make some further (small) edits to make the article work right as a standalone piece. If you're not sure what to do, please make a start and a reviewer can provide feedback.

Looking at the SIG node article you linked above as an example it looks pretty similar, I'm not sure what further changes might be needed (apart from header change which I posted now).

@xing-yang or @jsafrane  Can you please take a look again?

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 25]
```diff
@@ -0,0 +1,96 @@
+---
+layout: blog
+title: "Persistent Volume Last Phase Transition Time"
+date: 2023-09-20
```
Comment (xing-yang) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:4]: We need a new publication date?

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+slug: persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes v1.28, we have introduced a new alpha feature that aims to improve PersistentVolume (PV)
```
Comment (xing-yang) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:10]: In Kubernetes v1.28 -> In the recent Kubernetes v1.28 release

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+   For example, it is now trivial to write a script or a policy that deletes all PVs that have been in the `Released` phase for a week.
+
+2. Monitor Storage Health
+
+   By analyzing the phase transition times of PVs, administrators can monitor storage health more effectively.
+   For example, they can identify PVs that have been in the `Pending` phase for an unusually long time, which may indicate underlying issues with the storage provisioner.
+
+## How to use it
+
+The `lastPhaseTransitionTime` field is alpha for Kubernetes v1.28, so it requires
```
Comment (xing-yang) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:47]: The `lastPhaseTransitionTime` field is alpha starting from Kubernetes v1.28, ...

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 25 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: New date is set, is Oct 23 ok?

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 26 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: Changed.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 27 above)
Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: Changed.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 28]
```diff
@@ -0,0 +1,96 @@
+---
+layout: blog
+title: "Persistent Volume Last Phase Transition Time"
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:3]:
```suggestion
title: PersistentVolume Last Phase Transition Time in Kubernetes
```
or
```suggestion
title: PersistentVolume Last Phase Transition Time
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+date: 2023-10-23
+slug: persistent-volume-last-phase-transition-time
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In the recent Kubernetes v1.28 release, we have introduced a new alpha feature that aims to improve PersistentVolume (PV)
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:10]:
```suggestion
In the recent Kubernetes v1.28 release, we (SIG Storage) introduced a new alpha feature that aims to improve PersistentVolume (PV)
```

Please make it clear that _we_ is not Red Hat.

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 30]
```diff
@@ -0,0 +1,96 @@
...
+storage management and help cluster administrators gain better insights into the lifecycle of PVs.
+With the addition of the `lastPhaseTransitionTime` field into the status of a PV, administrators will be able to track the last time a PV transitioned to a different [phase](/docs/concepts/storage/persistent-volumes/#phase), allowing for more efficient and informed resource management.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:12]:
```suggestion
With the addition of the `lastPhaseTransitionTime` field into the status of a PV,
cluster administrators are now able to track the last time a PV transitioned to a different
[phase](/docs/concepts/storage/persistent-volumes/#phase), allowing for more efficient
and informed resource management.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 31]
```diff
@@ -0,0 +1,96 @@
...
+
+## Why do we need new PV field? {#why-new-field}
+
+PersistentVolumes in Kubernetes play a crucial role in providing storage resources to workloads running in the cluster.
+However, managing these PVs effectively can be challenging, especially when it comes to determining the last time a PV transitioned between different phases, such as `Pending`, `Bound` or `Released`.
+Administrators often need to know when a PV was last used or transitioned to certain phases, for instance, to implement retention policies, perform cleanup, or monitor storage health.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:18]:
```suggestion
However, managing these PVs effectively can be challenging, especially when it comes
to determining the last time a PV transitioned between different phases, such as
`Pending`, `Bound` or `Released`.
Administrators often need to know when a PV was last used or transitioned to certain
phases; for instance, to implement retention policies, perform cleanup, or monitor storage health.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 32]
```diff
@@ -0,0 +1,96 @@
...
+
+In the past, Kubernetes users have faced data loss issues when using the `Delete` retain policy and had to resort to the safer `Retain` policy.
+By introducing the `lastPhaseTransitionTime` field, we aim to provide a more generic solution that can be used for various use cases,
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:21]:
```suggestion
When we planned the work to introduce the new `lastPhaseTransitionTime` field, we
wanted to provide a more generic solution that can be used for various use cases,
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md [hunk 33]
```diff
@@ -0,0 +1,96 @@
...
+to beta and the release that graduates the field to general availability (GA). That means that
+the earliest release where this field could be generally available is Kubernetes 1.32,
+likely to be scheduled for early 2025.
+
+## Getting involved
+
+We always welcome new contributors so if you would like to get involved you can
+join our [Kubernetes Storage Special-Interest-Group](https://github.com/kubernetes/community/tree/master/sig-storage) (SIG).
+
+If you would like to share feedback, you can do so on our [public Slack channel](https://app.slack.com/client/T09NY5SBT/C09QZFCE5). If you're not already part of that Slack workspace, you can visit https://slack.k8s.io/ for an invitation.
```
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:85]:
```suggestion
If you would like to share feedback, you can do so on our
[public Slack channel](https://app.slack.com/client/T09NY5SBT/C09QZFCE5).
If you're not already part of that Slack workspace, you can visit https://slack.k8s.io/ for an invitation.
```

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md (hunk 25 above)
Comment (sftim) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md]: Sounds good, thanks.

**Review by sftim — COMMENTED** [review]
Hi. I've suggested some tweaks to make.

/lgtm cancel
/hold cancel

File: content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md
Comment (sftim) [inline content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:1]: Please change the filename to match the publication date (we don't yet have a lint for this, sorry).

Comment (RomanBednar) [reply content/en/blog/_posts/2023-08-15-pv-last-phase-transtition-time.md:1]: Sure, changed.

**Comment by tengqm** [comment]
Feedbacks addressed, publish date confirmed.
/lgtm
/approve

**Pull request merged**